File: python/selection.py
# Time: O(n^2)
# Space: O(1)

from time import time
from random import randint

raw_data = []

def random_generator(target):
    for i in range(target):
        raw_data.append(randint(0, target))
def selection():
    for i in raw_data:
        point = 0
        for j in range(len(raw_data) - 1):
            if raw_data[point] > raw_data[point+1]:
                raw_data[point], raw_data[point+1] = raw_data[point+1], raw_data[point]
            point += 1
        point += 1

random_generator(10)
print(f"Raw data: {raw_data}")
start = time()
selection()
finish = time()
print(f"Sorted data: {raw_data}")
print(f"Time: {finish-start}")

File: c/selection.c
#include <stdio.h>

int raw_data[] = {7,4,2,5};

void selection()   {
    for (int i = 0; i < sizeof(raw_data) / sizeof(raw_data[0]); i++)    {
        int target_point = 0;
        for (int j = 0; j < sizeof(raw_data) / sizeof(raw_data[0]) - 1; j++)    {
            if (raw_data[target_point] > raw_data[target_point + 1])    {
                int temp = raw_data[target_point + 1];
                raw_data[target_point + 1] = raw_data[target_point];
                raw_data[target_point] = temp; 
            }
            target_point++;
        }
    }
}

int main()  {
    printf("Raw data: ");
    for (int i = 0; i < sizeof(raw_data) / sizeof(raw_data[0]); i++)    {
        printf("%d ", raw_data[i]);
    }

    selection();

    printf("\nSorted data: ");
    for (int i = 0; i < sizeof(raw_data) / sizeof(raw_data[0]); i++)    {
        printf("%d ", raw_data[i]);
    }
    return 0;
}